pkgs/games/mindustry/default.nix

```diff
@@ -167,33 +150,26 @@ stdenv.mkDerivation rec {
   ];
   nativeBuildInputs = [
     pkg-config
-    gradleWithJdk
+    gradle
     makeWrapper
     jdk
   ] ++ lib.optionals enableClient [
     ant
     copyDesktopItems
+    curl
+    wget
   ];
 
   desktopItems = lib.optional enableClient desktopItem;
 
-  buildPhase = with lib; ''
-    export GRADLE_USER_HOME=$(mktemp -d)
+  gradleFlags = [ "-Pbuildversion=${buildVersion}" "-Dorg.gradle.java.home=${jdk}" ];
 
-    # point to offline repo
-    sed -ie "1ipluginManagement { repositories { maven { url '${deps}' } } }; " Mindustry/settings.gradle
-    sed -ie "s#mavenLocal()#mavenLocal(); maven { url '${deps}' }#g" Mindustry/build.gradle
-    sed -ie "s#mavenCentral()#mavenCentral(); maven { url '${deps}' }#g" Arc/build.gradle
-    sed -ie "s#wget.*freetype.* -O #cp ${freetypeSource} #" Arc/extensions/freetype/build.gradle
-    sed -ie "/curl.*glew/{;s#curl -o #cp ${glewSource} #;s# -L http.*\.zip##;}" Arc/backends/backend-sdl/build.gradle
-    sed -ie "/curl.*sdlmingw/{;s#curl -o #cp ${SDLmingwSource} #;s# -L http.*\.tar.gz##;}" Arc/backends/backend-sdl/build.gradle
-
-    pushd Mindustry
+  buildPhase = with lib; optionalString enableServer ''
+    gradle server:dist
   '' + optionalString enableClient ''
-
     pushd ../Arc
-    gradle --offline --no-daemon jnigenBuild -Pbuildversion=${buildVersion}
-    gradle --offline --no-daemon jnigenJarNativesDesktop -Pbuildversion=${buildVersion}
+    gradle jnigenBuild
+    gradle jnigenJarNativesDesktop
     glewlib=${lib.getLib selectedGlew}/lib/libGLEW.so
```

Comment:
OT but the rest of these feel like they should happen in install and/or fixup phases.

Reply:
dist:desktop and dist:server also build all of the subtasks, so I don't really think so